src/ci/scripts/should-skip-this.sh

```diff
@@ -18,7 +18,12 @@ BASE_COMMIT="$(git merge-base FETCH_HEAD HEAD)"
 
 echo "Searching for toolstate changes between $BASE_COMMIT and $(git rev-parse HEAD)"
 
-if git diff "$BASE_COMMIT" | grep --quiet "^index .* 160000"; then
+if (git log HEAD..."$BASE_COMMIT" --diff-filter=D --summary | grep delete \
+        | grep --quiet "compiler/rustc_error_codes/src/error_codes/"); then
+    echo "Error code explanations should never be removed!"
+    echo "Take a look at E0001 to see how to handle it."
+    exit 1
```

Comment:
Why is this check added to a skip script? This doesn't seem like the right place to add this.

I would probably try moving this to a script like validate-toolstate.sh in mingw-check, and hooking it up to run there.

Reply:
I simply didn't know where to put it. Now I know, thanks for the indication! :)